lib/controller/kontext.py

```diff
@@ -615,17 +615,21 @@ def _check_corpus_access(self, action_name, form, action_metadata):
 
         Returns: a 2-tuple (copus id, corpus variant)
         """
-        def validate_access(cn, allowed):
-            if cn and cn.lower() in allowed:
-                return True, allowed[cn.lower()]
+        def validate_access(cn, allowed, lowercase_names: bool):
+            if cn and lowercase_names:
+                cn = cn.lower()
+
+            if cn and cn in allowed:
+                return True, allowed[cn]
             else:
                 return False, ''
 
         with plugins.runtime.AUTH as auth:
             allowed_corpora = auth.permitted_corpora(self.session_get('user'))
             if not action_metadata['skip_corpus_init']:
                 corpname, redirect = self._determine_curr_corpus(form, allowed_corpora)
-                has_access, variant = validate_access(corpname, allowed_corpora)
+                lowercase_names = auth.lowercase_corpora_names
```

Comment:
This is not perfect. I would prefer if "auth" was accessible in "validate_access".

One way to achieve this would require defining "validate_access" under the context manager, which would look good.

Reply:
I'm even thinking about making `validate_access`  a part of the `AbstractAuth` interface. This would allow hiding all the suggested functionality inside the plug-in.